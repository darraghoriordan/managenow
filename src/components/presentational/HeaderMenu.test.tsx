import * as React from "react";
import { MemoryRouter } from "react-router";
import * as renderer from "react-test-renderer";
import HeaderMenu from "./HeaderMenu";

it("renders correctly for non-authenticated user", () => {
  const component = renderer.create(
    <MemoryRouter initialEntries={["/"]}>
      <HeaderMenu
        authenticated={false}
        // tslint:disable-next-line:jsx-no-lambda
        onSignOut={() => true}
      />
    </MemoryRouter>
  );
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});

it("renders correctly for authenticated user", () => {
  const component = renderer.create(
    <MemoryRouter initialEntries={["/"]}>
      <HeaderMenu
        authenticated={true}
        displayName="darragh"
        // tslint:disable-next-line:jsx-no-lambda
        onSignOut={() => true}
      />
    </MemoryRouter>
  );
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
