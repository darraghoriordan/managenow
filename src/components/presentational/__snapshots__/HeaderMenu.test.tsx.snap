// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly for authenticated user 1`] = `
<div
  className="ui top fixed menu"
>
  <div
    className="ui container"
  >
    <a
      className="header item"
      href="/"
      onClick={[Function]}
    >
      New Leader List
    </a>
    <a
      className="item"
      href="/"
      onClick={[Function]}
    >
      Home
    </a>
    <a
      className="item"
      href="/team"
      onClick={[Function]}
    >
      Your Team
    </a>
    <div
      className="right menu"
    >
      <div
        aria-busy={undefined}
        aria-disabled={undefined}
        aria-expanded={false}
        aria-multiselectable={undefined}
        className="ui item dropdown"
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
        role="listbox"
        tabIndex={0}
      >
        <div
          aria-live="polite"
          className="text"
          role="alert"
        >
          darragh
        </div>
        <i
          aria-hidden="true"
          className="dropdown icon"
          onClick={[Function]}
        />
        <div
          className="menu transition"
        >
          <div
            aria-checked={undefined}
            aria-disabled={undefined}
            aria-selected={undefined}
            className="item"
            onClick={[Function]}
            role="option"
          >
            <i
              aria-hidden="true"
              className="edit icon"
            />
            <span
              className="text"
            >
              Signout
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders correctly for non-authenticated user 1`] = `
<div
  className="ui top fixed menu"
>
  <div
    className="ui container"
  >
    <a
      className="header item"
      href="/"
      onClick={[Function]}
    >
      New Leader List
    </a>
    <a
      className="item"
      href="/"
      onClick={[Function]}
    >
      Home
    </a>
    <a
      className="item"
      href="/list"
      onClick={[Function]}
    >
      The App
    </a>
    <div
      className="right menu"
    >
      <a
        className="item"
        href="/signin"
        onClick={[Function]}
      >
        Sign in
      </a>
    </div>
  </div>
</div>
`;
